core/src/test/scala/org/apache/spark/io/ChunkedByteBufferSuite.scala

```diff
@@ -56,6 +56,15 @@ class ChunkedByteBufferSuite extends SparkFunSuite {
     assert(chunkedByteBuffer.getChunks().head.position() === 0)
   }
 
+  test("SPARK-24107: writeFully() write buffer which is larger than bufferWriteChunkSize") {
+    val bufferWriteChunkSize = Option(SparkEnv.get).map(_.conf.get(config.BUFFER_WRITE_CHUNK_SIZE))
+            .getOrElse(config.BUFFER_WRITE_CHUNK_SIZE.defaultValue.get).toInt
```

Comment:
How about setting this value via `spark.buffer.write.chunkSize`?

Reply:
Ok. I have added. Please check